### ISSUE TITLE
A MERGE Statement Must be Terminated by a Semi-colon" Error with MSSQLSERVER 

### ISSUE THREAD

**Issue by dtorres-srga** [liquibase/liquibase#3941]
### Search first

- [X] I searched and no similar issues were found

### Description

I'm having trouble using MSSQLSERVER (Linux container). I created a change set that includes a MERGE INTO statement, but when I try to run it, Liquibase returns an error message: "A MERGE statement must be terminated by a semi-colon (;)."

I have tried terminating my change set with a semi-colon both in line and on a new line. I am using a SQL-based change set. 

### Steps To Reproduce

- You can use the following docker-compose file to setup MSSQLSERVER and a liquibase container 

```
docker run -e 'ACCEPT_EULA=Y' -e 'MSSQL_SA_PASSWORD=Temp123@' -p 1433:1433 --name sql1 -h sql1 mcr.microsoft.com/mssql/server:2022-latest
```

- Create a test database 

- Run liquibase with the following changeset
```
--liquibase formatted sql

--changeset dtorres:0
CREATE TABLE Gender (
   Id INT PRIMARY KEY,
   Gender TEXT NOT NULL
);


--changeset dtorres:1 runOnChange:true
MERGE INTO Gender AS t
USING (
   VALUES
      (1, 'Male'),
      (2, 'Female'),
      (3, 'Non-Binary'),
      (4, 'Unknown')
) AS s (id, Gender)
ON t.id = s.id
WHEN MATCHED THEN
   UPDATE SET t.Gender = s.Gender
WHEN NOT MATCHED THEN
   INSERT (id, Gender) VALUES (s.id, s.Gender);
```

You will see the following error: 
```
Unexpected error running Liquibase: Migration failed for changeset changelog/changelog.sql::1::dtorres:
liquibase2_1  |      Reason: liquibase.exception.DatabaseException: A MERGE statement must be terminated by a semi-colon (;). [Failed SQL: (10713) MERGE INTO Gender AS t
liquibase2_1  | USING (
liquibase2_1  |    VALUES
liquibase2_1  |       (1, 'Male'),
liquibase2_1  |       (2, 'Female'),
liquibase2_1  |       (3, 'Non-Binary'),
liquibase2_1  |       (4, 'Unknown')
liquibase2_1  | ) AS s (id, Gender)
liquibase2_1  | ON t.id = s.id
liquibase2_1  | WHEN MATCHED THEN
liquibase2_1  |    UPDATE SET t.Gender = s.Gender
liquibase2_1  | WHEN NOT MATCHED THEN
liquibase2_1  |    INSERT (id, Gender) VALUES (s.id, s.Gender)]
liquibase2_1  | 
liquibase2_1  | For more information, please use the --log-level flag
```

### Expected/Desired Behavior

changeset should be applied successfully.

### Liquibase Version

4.20.0

### Database Vendor & Version

Microsoft SQL Server 2022 (RTM-CU1) (KB5022375) - 16.0.4003.1 (X64)  (LINUX CONTAINER)

### Liquibase Integration

CLI (Docker container)

### OS and/or Infrastructure Type/Provider

Docker containers

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by dtorres-srga** [comment]
Note, I did find a **workaround** (actually I got it from your discord) 

Set `splitStatements` to be false in your change set. Then you just need to ensure that your merge statement is terminated with a semi-colon.

**Comment by FBurguer** [comment]
Confirmed to be an issue for liquibase 4.20. Thanks for the submission. If you can submit a pr with a PR to fix this, it would  help speeding up the process. Thanks!

**Comment by filipelautert** [comment]
Hello @dtorres-srga ! I managed to reproduce your error using the example on Liquibase `4.20.0 `and `4.21.1 `. But it has been fixed on version 4.23.0:

```
Starting Liquibase at 14:50:33 (version 4.23.0 #10709 built at 2023-06-23 21:18+0000)
Liquibase Version: 4.23.0
Liquibase Open Source 4.23.0 by Liquibase

Running Changeset: 3941/changelog.sql::0::dtorres
Running Changeset: 3941/changelog.sql::1::dtorres

UPDATE SUMMARY
Run:                          2
Previously run:               0
Filtered out:                 0
-------------------------------
Total change sets:            2

Liquibase: Update has been successful.
Liquibase command 'update' was executed successfully.
```


Seems it was fixed here -> https://github.com/liquibase/liquibase/pull/4202 . 

Could you confirm that it works for you?
